raiden/tests/smart_contracts/test_decoder_netting_channel.py

```diff
@@ -149,3 +152,94 @@ def test_decode_refund_transfer(
     refund_transfer.sign(PrivateKey(privatekey0, ctx=GLOBAL_CTX, raw=True), address0)
 
     assert_decoder_results(refund_transfer, decoder)
+
+
+def test_decode_tampered_direct_transfer(tester_state, tester_nettingchannel_library_address):
+    privatekey0 = tester.DEFAULT_KEY
+    address0 = privatekey_to_address(privatekey0)
+
+    decoder = deploy_decoder_tester(tester_state, tester_nettingchannel_library_address)
+
+    direct_transfer = make_direct_transfer()
+    direct_transfer.sign(PrivateKey(privatekey0, ctx=GLOBAL_CTX, raw=True), address0)
+
+    message_encoded = direct_transfer.encode()
+    transfer_raw, _ = decoder.getTransferRawAddress(message_encoded)
+
+    names_slices = compute_slices(messages.DirectTransfer.fields_spec)
+    for name, slice_ in names_slices.iteritems():
+        if name == 'signature':
+            continue
+
+        tampered_transfer = bytearray(transfer_raw)
```

Comment:
Can you add a tiny comment above explaining what's happening in these loops?

"Pop the first and last byte of each field and make sure decoding fails" or something along those lines.

Also since this is repeating here and also in the other test below perhaps put it in a helper function and move the aforementioned comment there?